core/tests/player_test.py

```diff
@@ -60,5 +61,30 @@ def test_add_species__removes_discard_card_from_player_hand_cards(self):
             self.player.hand_cards
         )
 
-    # Todo: write add_species position left and right tests.
+    def test_add_species__adds_species_to_the_left_of_existing_species(self):
```

Comment:
Did you mean to add two underscores in the name?

Reply:
Yeah. I followed the approach of using two underscores to separate: a) the end of the function name that the test is testing; from b) a description of the test. Would you prefer that we drop this approach?

Reply:
Having too-long function names can end up making the code harder to digest/read than the opposite (which is usually to make it clearer). Normally I would suggest comments would also be OK, and given that a test may actually test one theme via calling potentially more than one function, I feel like we don't need to feel bound to putting the function name explicitly in the test name if it's clearer to do something else.

Or another way of looking at it -- it seem's unnatural to have "add_species" immediately followed by "adds_species" in the test name. I would suggest we either try and align the test names with the function(s) that are being tested (with comments for clarity), or make them descriptive of the test, but not both. What do you think about that as a suggestion?

Also fine to go with it, as-is though too.

Reply:
Nice comment; I agree. I suggest going for the second approach of function name followed by description in comment.